### PR TITLE
Drop jQuery

### PR DESCRIPTION
**Pull request by fregante**
- Replaces https://github.com/OctoLinker/OctoLinker/pull/1049

Untested, basically a manual search and replace:

- if a single element was expected, I used `querySelector` + `?.`
- if multiple elements were expected, I used `querySelectorAll` in a wrapper that produces an array
- use $() and $$() match the console utilities: https://developer.chrome.com/docs/devtools/console/utilities/#querySelector-function

### CODE REVIEW

**Comment by stefanbuck** [comment]
This is great 🎉 I'll take a closer look at your changes over the holidays. Just one note on the failing test. Sometime `jsdom` is not behaving as expect and I think for some reason `querySelectorAll` is not available. I can dig into this as part of the review.

**Comment by fregante** [comment]
Ok it seems to work now, but I probably messed something up because the tests are still failing.

**Comment by xt0rted** [comment]
@fregante try rebasing now that the tests & dependencies were updated.

File: packages/blob-reader/helper.js [hunk 1]
```diff
@@ -1,19 +1,24 @@
...
-function getBlobWrapper(rootElement = document) {
-  const ret = [
-    ...[].slice.call(rootElement.getElementsByClassName('blob-wrapper')),
-    ...[].slice.call(rootElement.getElementsByClassName('js-blob-wrapper')),
-    ...[].slice.call(
-      rootElement.querySelectorAll('[class*="highlight-source-"]'),
-    ),
-  ];
+function getBlobCodeInner(el) {
+  return $$('.blob-code-inner');
```
Comment (xt0rted) [inline packages/blob-reader/helper.js:10]: Should `el` be passed to `$$()`?

File: packages/blob-reader/helper.js [hunk 2]
```diff
@@ -102,11 +105,16 @@ function getLineNumber(el) {
   }
 
   // split diff view
-  let lineNumber = $(el).closest('td').prev().data('line-number');
+  let lineNumber = el
+    .closest('td')
+    ?.previousElementSibling?.getAttribute('line-number');
```
Comment (xt0rted) [inline packages/blob-reader/helper.js:110]:
```suggestion
    ?.previousElementSibling?.getAttribute('data-line-number');
```

File: packages/blob-reader/helper.js [hunk 3]
```diff
@@ -102,11 +105,16 @@ function getLineNumber(el) {
...
 
   // unified diff view
   if (!lineNumber) {
-    lineNumber = $(el).closest('tr').find('td').data('line-number');
+    lineNumber = el
+      .closest('tr')
+      ?.querySelector('td')
+      ?.getAttribute('line-number');
```
Comment (xt0rted) [inline packages/blob-reader/helper.js:117]:
```suggestion
      ?.getAttribute('data-line-number');
```

**Review by xt0rted — CHANGES_REQUESTED** [review]
I was able to track down why this isn't working. I left comments on the spots I could but there's one more spot I can't since it wasn't edited as part of this PR.

https://github.com/OctoLinker/OctoLinker/blob/a93670e0922a0a96a8224622366ba44ff152de9e/packages/blob-reader/helper.js#L120-L124

This needs to change to:

```js
if (lineNumber) {
  if (!Number.isNaN(lineNumber)) {
    return parseInt(lineNumber, 10);
  }
}
```

jQuery was returning the data attribute values as numbers, but `getAttribute()` returns a string and there's code downstream that needs them as a number.

File: packages/blob-reader/helper.js (hunk 1 above)
Comment (fregante) [reply packages/blob-reader/helper.js]: The perils of _not_ using every lint rule available 😃

**Comment by fregante** [comment]
Boom, tests pass

**Comment by stefanbuck** [comment]
I did a couple of manual spot checks and seems to work. Really great work from both of your.

**Pull request merged**